lib/blocktree/blocktree.go

```diff
@@ -68,6 +70,50 @@ func (bt *BlockTree) GenesisHash() Hash {
 	return bt.head.hash
 }
 
+// GetAllHashesForParentDepth will get all hashes with the same dept for a given header
+func (bt *BlockTree) GetAllHashesForParentDepth(header *types.Header) (map[common.Hash]*big.Int, error) {
+	hashes := map[common.Hash]*big.Int{}
+
+	parent := bt.getNode(header.ParentHash)
+	if parent == nil {
+		err := fmt.Errorf("cannot find parent block in blocktree")
+		log.Error("Could not getNode for ParentHash", "header.ParentHash", header.ParentHash, "err", err)
+		return hashes, err
+	}
+
+	for _, child := range bt.head.children {
+		if child.depth.Cmp(parent.depth) == -1 {
+			log.Debug("breaking range bt.head.children", "child.hash", child.hash.String(), "child.depth", child.depth)
+			break
+		} else if child.depth.Cmp(parent.depth) == 0 {
+			if hashes[child.hash] != nil {
+				log.Debug("adding child.hash", "child.hash", child.hash.String(), "child.depth", child.depth)
+				hashes[child.hash] = parent.depth
+			}
+		}
+
+		hashes = bt.getHashes(child, hashes)
+	}
+	return hashes, nil
+}
+
+func (bt *BlockTree) getHashes(n *node, hashes map[common.Hash]*big.Int) map[common.Hash]*big.Int {
```

Comment:
I'm not sure what this is doing, but you don't need to return a map, an array of hashes is all you need. since all the nodes should be at the same depth

Reply:
outdated